packages/graphql-language-service-server/src/GraphQLCache.ts

```diff
@@ -90,7 +90,7 @@ export class GraphQLCache implements GraphQLCacheInterface {
     this._fragmentDefinitionsCache = new Map();
     this._typeDefinitionsCache = new Map();
     this._typeExtensionMap = new Map();
-    this._parser = parser || parseDocument;
+    this._parser = parser;
```

Comment:
I was thinking that the default should be set here, rather than in `MessageProcessor`

Reply:
Looking at it, I'm not sure that makes sense. `addHandlers` is instantiating `MessageProcessor`, which in turn is instantiating the `GraphQLCache`. But `MessageProcessor` is also using the parser, so it needs a value set, which means it needs to set a default, or force providing a parser.